gnome-initial-setup/pages/display/gis-display-page.c

```diff
@@ -72,41 +70,57 @@ set_toggle_options_from_config (GisDisplayPage *page)
 }
 
 static gboolean
-read_screen_config (GisDisplayPage *page)
+should_display_overscan (GisDisplayPage *page)
 {
   GisDisplayPagePrivate *priv = gis_display_page_get_instance_private (page);
-  GnomeRRConfig *current;
-  GnomeRROutputInfo **outputs;
-  GnomeRROutputInfo *output;
-  int i;
 
-  gnome_rr_screen_refresh (priv->screen, NULL);
+  if (!priv->current_output)
+    return FALSE;
+
+  return cc_display_monitor_supports_underscanning (priv->current_output);
+}
+
+static gboolean
+read_screen_config (GisDisplayPage *page)
+{
+  GisDisplayPagePrivate *priv = gis_display_page_get_instance_private (page);
+  CcDisplayConfig *current;
+  GList *outputs, *l;
+  gboolean visible = FALSE;
 
   g_clear_object (&priv->current_config);
+  priv->current_output = NULL;
 
-  current = gnome_rr_config_new_current (priv->screen, NULL);
-  gnome_rr_config_ensure_primary (current);
-  priv->current_config = current;
+  current = cc_display_config_manager_get_current (priv->manager);
+  if (!current)
+    {
+      g_critical ("Could not get primary output information. Hiding overscan page.");
+      return FALSE;
```

Comment:
I don't think we'd hide it from this code path -- the call to `gtk_widget_set_visible()` is at the end of this function.

Reply:
Oops, right! This was an overlook while refactoring. I'll change it to make sure we `goto` the end in this case and set visible accordingly.

Reply:
...or better: move the code that sets the visibility out of this function and into where it used to be before: `gis_display_page_constructed`

Reply:
Actually, I can't move it exactly there because of the asynchronous initialization of the CcDisplayManager, so scrap that.